### ISSUE TITLE
ANSI escape character sequences are printed  in Windows cmd.exe when outputting to stderr

### ISSUE THREAD

**Issue by lpinner** [jd/daiquiri#17]
I'm stuck using Windows :(for work).  

The ANSI escape character sequences are printed  in Windows cmd.exe when outputting to stderr.

```
#test.py
import daiquiri
daiquiri.setup()
logger = daiquiri.getLogger()
logger.error("something went wrong")
```
![test](https://user-images.githubusercontent.com/4827816/27999854-ceb64e56-6564-11e7-8c0f-44d15b8980d7.png)
I can workaround using [colorama](https://pypi.python.org/pypi/colorama) (the library that IPython uses).

```
#test_colour.py
import colorama 
colorama.init()

import daiquiri
daiquiri.setup()
logger = daiquiri.getLogger()
logger.error("something went wrong")
```
![test_colour](https://user-images.githubusercontent.com/4827816/28001334-14d12596-656f-11e7-8f39-4ff450492f37.png)

Colours also work when script is run using ipython as colorama is being used automatically (i.e. `ipython test.py`).

Is there a way to tell daiquiri not to output colour escape chars?  
Even better ; ), would you consider using colorama as an optional extra?

**Comment by jd** [comment]
Oh we should definitely switch to colorama. @blakev maybe you should take a look into that for https://github.com/jd/daiquiri/pull/9 ?

**Comment by blakev** [comment]
For sure, if you're not against adding the dependency.

**Comment by jd** [comment]
No, all good for me. Worst thing we could make it optional, but well. It's a common use case.

**Comment by eli-b** [comment]
Once colorama is used, you can document that Windows is supported in `setup.cfg`.

**Comment by jd** [comment]
Still open, and colorama is still the plan.

Where things stand: it never landed. #9 was closed unmerged in 2020, `pyproject.toml` depends on `python-json-logger` and nothing else, and CI runs on ubuntu only, so nobody here exercises the Windows path. The package metadata still claims `Operating System :: POSIX :: Linux`.

What has changed since 2017 is that the escape sequences do work on current Windows consoles. Windows Terminal, the default host on Windows 11, handles them, and conhost does too once a process turns on virtual terminal processing. Legacy `cmd.exe` in a plain conhost window still prints them raw, which is the screenshot above. daiquiri does nothing to tell those cases apart, so the behaviour is entirely down to which console you happen to be in.

The fix is the one @lpinner asked for: colorama as an optional extra, with `colorama.just_fix_windows_console()` called at setup time on win32. The Windows classifier can follow once CI actually covers it.

Tagging help wanted. A patch from someone who can verify it on a real Windows box will get this done faster than I will.
